applications/asset_tracker_v2/src/main.c

```diff
@@ -119,6 +119,16 @@ static struct module_data self = {
 	.supports_shutdown = true,
 };
 
+NRF_MODEM_LIB_ON_INIT(asset_tracker_init_hook, on_modem_lib_init, NULL);
+
+/* Initialized to value different than success (0) */
+static int modem_lib_init_result = -1;
+
+static void on_modem_lib_init(int ret, void *ctx)
+{
```

Comment:
Will this work for the scenario of modem firmware FOTA?
There is an query of intermediate modem init result so as to do a reset to complete FOTA.